connectors/source.py

```diff
@@ -239,7 +260,7 @@ def __init__(self, config):
 
     def set_defaults(self, default_config):
         for name, item in default_config.items():
-            self._defaults[name] = item["value"]
+            self._defaults[name] = item.get("value", None)
```

Comment:
Is this line needed?

Reply:
I was worried that not specifying a `value` would cause an error here, but it doesn't. I removed it and added some tests to cover that.

Reply:
Since it's internal, I think it's okay - we can address it later